Flepper.QueryBuilder/Utils/ParameterObjectBuilder.cs

```diff
@@ -41,13 +42,32 @@ internal static Type CreateClass(IDictionary<string, object> parameters, string
                 return Types[className];
 
             var typeBuilder = ModuleBuilder.DefineType(className ?? Guid.NewGuid().ToString(), TypeAttributes.Public | TypeAttributes.Class | TypeAttributes.AutoClass | TypeAttributes.AnsiClass | TypeAttributes.BeforeFieldInit | TypeAttributes.AutoLayout, null);
-            typeBuilder.DefineDefaultConstructor(MethodAttributes.Public | MethodAttributes.SpecialName | MethodAttributes.RTSpecialName);
-            foreach (var parameter in parameters)
-                CreateProperty(typeBuilder, parameter.Key.Replace("@", ""), parameter.Value is null ? typeof(object) : parameter.Value.GetType());
+            typeBuilder.DefineDefaultConstructor(MethodAttributes.Public | MethodAttributes.SpecialName | MethodAttributes.RTSpecialName);           
+            foreach (var parameter in parameters)            
+                CreateProperty(typeBuilder, parameter.Key.Replace("@", ""), GetTypeParameter(parameter));            
             var type = typeBuilder.CreateTypeInfo().AsType();
             Types.TryAdd(type.FullName, type);
             return type;
         }
+        
+        internal static bool IsIQueryBuilderParameter(object value)
```

Comment:
When using C# pattern matching in other methods, you must remove this one

Reply:
Okay